src/test/pthreads/CMakeLists.txt

```diff
@@ -1,19 +1,31 @@
-include_directories(${RT_INCLUDES})
+## register the tests
+add_test(NAME pthreads COMMAND ../target/debug/test_pthreads --libc-passing)
+list(APPEND TestNames pthreads)
 
-## create and install an executable that can run outside of shadow
-add_executable(test-pthreads test_pthreads.c)
+## FIXME:
+##    shadow currently doesn't handle calls to malloc/free, can either:
+##      - switch to mmap/munmap in test_pthreads plugin
+##      - enable pthreads-shadow-ptrace when/if malloc/free implemented in shadow
```

Comment:
I'm not clear on this. Why does Shadow need to handle calls to malloc/free?

Reply:
If dynamic allocation is kept in this test, Shadow needs to support `malloc/alloca` for the test to run under Shadow. Currently, the test is only run outside.

Reply:
Shadow does support `malloc()`, and I haven't tried `alloca()`, but I don't think it uses a syscall so it should work.

If there is a problem with dynamic allocation in Shadow, please make a bug report with an example because that's definitely something we would need to fix.

Reply:
Removed all the calls to `libc::malloc/free`, and now the tests pass under `shadow-plugin` and `shadow-preload`. Thanks for the tips!